### PR TITLE
docs: Add `variant` and `css` docs for variance [WEB-1413]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Initial pass at docs for new `variant` and `css` functions.  These are work in progress.

* Adds basic documentation for the `variant` and `css` documentation for variance API in place of original gamut-styles documention.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-1413
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607dc2661468c11128928a1b--gamut-preview.netlify.app>https://607dc2661468c11128928a1b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607dc2661468c11128928a1b>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🎉

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 1]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
 
-There may be cases where you want to change many props at the same time to achieve a different effect.
-Coordinating these different prop variations can be a bit frustrating to do manually, in both typescript and plain javascript.
-To make this easier we have **`variant`**, a function that is meant to make creating these combinations easy and scalable.
+## `css`
 
-```tsx
-import styled from '@emotion/styled';
-import { variant } from '@codecademy/gamut-styles';
+A standalone theme aware method that accepts and object of system props and returns the correct style object. Is cached after being called with theme once, preventing more expensive calculations.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:16]:
```suggestion
A standalone theme aware method that accepts an object of system props and returns the correct style object. Is cached after being called with theme once, preventing more expensive calculations.
```

Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:16]:
```suggestion
A standalone theme aware method that accepts an object of system props and returns the correct style object. It is cached after being called with theme once, preventing more expensive calculations.
```

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 2]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
-Using your variants:
+## `variant`
 
-```tsx
-<Badge variant="primary" />
+Built on top of `css` that accepts a number of mutually exclusive states a component can be in. Options include:
 
-<Badge variant="secondary" />
-```
+- `prop`: The prop key that variants should be. Defaults to `variant`
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:34]: Nit: every other item in this list ends with one, heh

```suggestion
- `prop`: The prop key that variants should be. Defaults to `variant`.
```

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 3]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
-    primary: { color: 'blue' },
-    secondary: { color: 'green' },
-  },
-});
-
-const Badge = styled.p(colorVariant)
+import { variant } from '@codecademy/gamut-styles';
 
-<Badge />; // color: blue
+const Anchor = styled.a(variant({
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:46]: Nittiest of the nites: `css({` was on a new line above, but this one is on the same line? Maybe standardize to make it easier to skim through the docs?

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 4]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
+   defaultVariant: 'interface',
+   variants: {
+       interface: {
+           color: 'text',
+       },
+       inline: {
+           color: 'primary',
+        },
+    }
+});
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:56]: Can we file tickets somewhere about these two parsing bugs? lol

Screenshot of a different font for the a link and blue color for interface

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 5]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
+## Property Passthrough
 
-```tsx
-const colorVariant = variant({
-  prop: 'colorVariant',
-  variants: {
-    primary: { color: 'blue' },
-  },
-});
+This change lets us configure anything that is likely to be non dynamic but required for a complete and discrete style. The typings for both `variant` and `css` accept all properties that are not present on the passed configuration.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:61]: > This change lets us configure anything that is likely to be non dynamic but required for a complete and discrete style.

I don't understand what this means 😕 ... could you maybe rephrase in this thread so we can wordsmith it?

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 6]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
+const styled.ul(variant({
+   key: 'listVariant',
+   variants: {
+      vertical: {
+         listStyle: 'none',  // not a system prop but valid
+         p: 0,
+         m: 0,
+      },
+      emoji: {
+         listStyleType: "\1F44D",  // not
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:77]: I don't think it's clear from reading through this that `// not` means TypeScript won't allow it (or what "will not accept values" means). Maybe include an abridged TypeScript error in the comment?

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx [hunk 7]
```diff
@@ -11,93 +11,122 @@ import { Meta } from '@storybook/addon-docs/blocks';
...
+         pr: [12, 24],
+         m: 0
+      }
+}));
 ```
 
-### Composing Variants
+When generating the styles the function will ignore the all props not specified by our configuration and simply pass them through to the result CSSObject.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/System/Variants.stories.mdx:84]: Can you show an example?

Also, personal pet peeve, nothing is ever "simple" - so words like "simply" shouldn't be in docs IMO.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A few things to think about - I think some of them are relevant for the last third of docs too?

File: packages/styleguide/stories/Foundations/System/Variants.stories.mdx (hunk 6 above)
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/System/Variants.stories.mdx:77]: Heh this I think is just a typo, may have deleted it accidentally.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.1-alpha.4cced8.0<br/>@codecademy/gamut-kit@0.3.33-alpha.4cced8.0<br/>@codecademy/gamut-labs@13.5.10-alpha.4cced8.0<br/>@codecademy/gamut-styles@9.5.2-alpha.4cced8.0<br/>@codecademy/gamut-tests@2.3.13-alpha.4cced8.0<br/>@codecademy/gamut@31.0.4-alpha.4cced8.0<br/>@codecademy/konamimojisplosion@0.2.13-alpha.4cced8.0<br/>@codecademy/markdown-overrides@0.5.25-alpha.4cced8.0<br/>@codecademy/styleguide@35.0.7-alpha.4cced8.0<br/>@codecademy/variance@0.9.1-alpha.4cced8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60800f009af319f40d4c8d5d--gamut-preview.netlify.app>https://60800f009af319f40d4c8d5d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60800f009af319f40d4c8d5d>Deploy Logs</a></p>

**Comment by codecaaron** [comment]
Closing in favor of https://github.com/Codecademy/client-modules/pull/1783
